knowledge-base/google-github-actions/auth/action-security.yml

```diff
@@ -0,0 +1,10 @@
+name: Authenticate to Google Cloud
+github-token:
+  environment-variable-name: ACTIONS_ID_TOKEN_REQUEST_TOKEN
+  is-default: false
+  permissions:
+    id-token: write
+    id-token-if: ${{ contains(with, 'token_format') && with['token_format'] == 'id_token' }}
```

Comment:
@h0x0er I am not sure if `token_format` is the right thing to look for. In some examples, token_format is not set, but `id_token` is set to `write`. https://github.com/google-github-actions/auth#configuring-gcloud

Reply:
@varunsh-coder thanks for the review. I reanalyzed it.. I just figure that `workload_identity_provider` is right candidate to check for.. As if you look [here](https://github.com/google-github-actions/auth/blob/48c46e6a594e8bd0b0b4806155ace5cad3aa8e24/src/main.ts#L92), `ACTIONS_ID_TOKEN_REQUEST_TOKEN` token is used only `workload_identity_provider` is set..

Reply:
@h0x0er yes, this seems fine. Can you please rebase? For some reason `GITHUB_TOKEN` is also getting rate limited, so I changed to using a PAT...So the test.yml is different now from when you created the fork...

Reply:
@varunsh-coder Will do that..